airflow/models/connection.py

```diff
@@ -308,7 +311,7 @@ def get_hook(self):
                 "Could not import %s when discovering %s %s", hook_class_name, hook_name, package_name
             )
             raise
-        return hook_class(**{conn_id_param: self.conn_id})
+        return hook_class(**{conn_id_param: self.conn_id}, **hook_params)
```

Comment:
```suggestion
        return hook_class(conn_id_param=self.conn_id, **hook_params)
```

Maybe I'm missing something but I don't understand the usage of dictionary unpacking for a predetermined dictionary here 🤔

Reply:
Nevermind, I just realized that `conn_id_param` refers to a variable, not a constant. Neat!

Reply:
I just tried to make fewer changes, sir. In my opinion your changes are more elegant :)

Reply:
I think my suggested changes are actually incorrect - `conn_id_param` is a variable, not a `"conn_id_param"` string.

The dict unpacking allows the parameter passed to `hook_class()` to change, since the Hooks all take different parameters to define the connection they should be using (`gcp_conn_id`, `aws_conn_id`, `airbyte_conn_id`, etc).

Sorry for the confusion, but you should ignore my initial suggestion and return this code to its original state 😄

Reply:
Yeap, I get it also tonight. Sorry about that. Now looks correct